### PR TITLE
Covert TabsAdapter.java to Kotlin

### LINKED ISSUE #2402: Convert TabsAdapter to kotlin
Is your feature request related to a problem? Please describe.

TabsAdapter is in Java

Describe the solution you'd like

Convert it in Kotlin

Additional context

Parent issue #1826

### PR DESCRIPTION
**Pull request by abdulwd**
Fixes #2402 

**Screenshots**

### CODE REVIEW

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+import org.kiwix.kiwixmobile.core.utils.StyleUtils.fromHtml
+
+class TabsAdapter internal constructor(
+  private val activity: AppCompatActivity,
+  private val webViews: List<KiwixWebView>,
+  private val painter: NightModeViewPainter
+) : RecyclerView.Adapter<TabsAdapter.ViewHolder>() {
+  private var listener: TabClickListener? = null
+  var selected = 0
+  @SuppressLint("ResourceType") override fun onCreateViewHolder(
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:49]: its kind of a lot to ask but do you feel up to converting this to xml? So much effort for just a Textview and 2 ImageView in a CardView.

Ah I guess

```
ConstraintLayout.LayoutParams(
        activity.getWindowWidth() / 2,
        -activity.getToolbarHeight() / 2 + activity.getWindowHeight() / 2
      )
```
this is the kicker. I imagine we could affect something very similar with a custom LayoutManager but I'll admit to never having written one. Eh ignore this comment and we can just think about the future of this class later on

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    parent: ViewGroup,
+    viewType: Int
+  ): ViewHolder {
+    val context = parent.context
+    val margin16 = context.resources.getDimensionPixelSize(R.dimen.activity_horizontal_margin)
+    val contentImage = ImageView(context)
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:55]: use `apply` when you are configuring objects. It should make this code a lot prettier as the nodes will be more structurally visible

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    contentImage.id = 1
+    contentImage.scaleType = ImageView.ScaleType.FIT_XY
+    val close = ImageView(context)
+    close.id = 2
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:59]: Some old knowledge is saying you shouldn't manually set IDs like this, I am half remembering a `generateId` function maybe on ViewCompat? Or was it just available on newer APIs so I wrote my own fake ViewCompat? Or we can declare ids in a resources file.

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    close.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.ic_clear_white_24dp))
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:60]: we may have an extension function for this

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+  interface TabClickListener {
+    fun onSelectTab(view: View, position: Int)
+    fun onCloseTab(view: View, position: Int)
+  }
+
+  class ViewHolder(view: View, val content: ImageView, val title: TextView, val close: ImageView) :
+    RecyclerView.ViewHolder(view)
+
+  init {
```
Comment (macgills) [inline core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:167]: should be at the top

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt (hunk 1 above)
Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:49]: When I was writing this a couple of years back I was feeling so cool. But now it feels so bad. XD

You can share if you have something in mind to get rid of this.

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt (hunk 5 above)
Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt]: https://github.com/kiwix/kiwix-android/pull/2452/commits/ac979a2534bac443b4fb2dc2f10a4542d994f5b9

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt (hunk 4 above)
Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt]: https://github.com/kiwix/kiwix-android/pull/2452/commits/ac979a2534bac443b4fb2dc2f10a4542d994f5b9

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt (hunk 2 above)
Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:65]: https://github.com/kiwix/kiwix-android/pull/2452/commits/ac979a2534bac443b4fb2dc2f10a4542d994f5b9

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt (hunk 3 above)
Comment (abdulwd) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt]: https://github.com/kiwix/kiwix-android/pull/2452/commits/5528a00ba3cdfc9207d8ead24fdebe7db5b9febb

File: core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt (hunk 1 above)
Comment (macgills) [reply core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt:59]: If you want to create a ticket we can discuss there but for now this works so I don't feel a pressing urge to poke it

**Review by macgills — APPROVED** [review]

**Pull request merged**